### DISCUSSION TITLE
🧑🏾‍🚀 Portfolio Module Review and Iterate

### DISCUSSION THREAD

**Discussion by SallyMcGrath** [CodeYourFuture/curriculum#386] [Ideas]
https://curriculum.codeyourfuture.io/portfolio/

If you've just run a portfolio module, please come and share your insights, ideas: what worked and what failed. What was absolutely essential and what can we skip.

Remember a portfolio module needs to be re-runnable, so people need to be able to do it two or more times without getting bored, so it can't be as prescriptive as earlier modules. But we do need to capture experience and insights to share across the community. 

Get stuck in!

**Comment by SallyMcGrath** [comment]
@ellietms reminding you about this thread.

**Reply by haldenb** [reply]
My thoughts on Londons first PF module

https://www.loom.com/share/ebeda48f7ccf4783887cefbaf51ca02f
https://www.loom.com/share/0a1979fadf704616a70a9315aa7c0b8f
https://docs.google.com/presentation/d/1IJsbhfNCQyeA-RmjnioEqcFl46wfUMyl7Mldfen6Ltw/edit#slide=id.g1ec32b8bef7_0_14

**Comment by alismith-dev** [comment]
I think my main bit of feedback from attending a (part of) a class with LDN9 was that it felt like the trainees were quite focused on demoing at the end of the module. Their main concern was shipping something to make the demo impressive.

(I should say that I totally understand how we ended up here - a) we _do_ want to focus Launch projects around the graduation/job fair demo, so it's natural to follow this model and b) having to present a demo a very good motivator. Based on Halden's feedback above, I think the latter was especially important for LDN9. Hopefully it will be less important for other cohorts.)

While I think a demo is a useful hook to frame the problem around, I worry that we're spending a fair bit of effort on demos that ultimately are unlikely to get them a job, and perhaps not enough effort on building a portfolio that is much more positive employment-wise. Based on my experience of reviewing junior candidates, a lot of paid bootcamp graduates have good looking, easy to understand and impressive portfolios - it clearly is a focus of their course. Whereas CYF graduates usually only have a Github profile that is full of projects named "JS2 Week 1" with no context on what this means making it difficult assess what has been achieved.

So going forward I'd suggest that we put a bit more emphasis on building a interesting project and then documenting what they built and learned on their portfolios. I think we can probably write some guidelines around this in the curriculum (maybe backed up by a workshop?) Another idea is that if we do still want to do a demo, we could record the demos, then get trainees to put them on YouTube and link them from the portfolios. This is a great way to give context to potential employers.

Finally, I worry that trainees (and volunteers, but perhaps this is a separate problem) don't perceive a poor portfolio as a problem and we need to do more to motivate the module. I wonder if we could do some kind of workshop around reviewing different portfolios to give context on what employers are looking for and identify anti-patterns.

**Reply by ellietms** [reply]
Hi @40thieves, I appreciate your feedback. 
We really did not focus on the demo and that session was a session before the demo day.  in the portfolio module, we focused on different things not just projects or demos. that group specifically had a very good motivated trainee and they did lots of features for their projects and they will put it in their portfolio now. I am not sure you have seen what they built but this is the link to their website:
https://song-sieve.vercel.app/ 

I think we need to actually give them this feeling this module is serious and they have to take it seriously. This group that they had done an amazing job admitted that most of the features they finished in the last week for the demo. 

I think based on what each individual is lacking either portfolio website, CV or projects we can help them.

**Comment by ellietms** [comment]
Thanks, @SallyMcGrath  for reminding me. I could not come back to this earlier.
I am going to write my opinion based on my experience with the first portfolio module: 

**Issues** :

1. We did not know how we could be helpful to all trainees who want to join this module and they are in different stages and different levels of their journey. I am going to be more specific about this: 

-  We had trainees who had a portfolio website and their CV was ready but their CV was not good enough for applying for jobs they believed their CV and portfolio were perfect.
- We had trainees who had good CVs but they did not have a portfolio website and also they needed to work on their CV too.
- We had trainees who had good portfolio websites and good CVs and they just wanted to work on their technical skills like making projects in a group or individually. They wanted to be part of the portfolio module because of all the workshops we had for CV, Linkedin and applying for jobs and the feedback was they found it helpful.
- We also had trainees who did not have a portfolio website but also did not want to commit to this module and it was challenging to help them because they were disappointed by everything!

**Issues I was facing:**

1. How I can lead this module to help everyone they can achieve a project, a portfolio website or a good CV and a good LinkedIn account at the end.
2. How I can group people based on their different levels and stages they are in and their commitment and availability as well.
3. Lack of support for the first two weeks.
4. Lack of commitments from trainees also made it sometimes difficult but I also wanted them to not feel any pressure and do it for themselves.
5. How many technical and PD volunteers do I need?
6. Lack of resources like a syllabus or some kind of path that I can follow and help me
7. What kind of subjects I can pick that can help everyone in their journey for workshops 
8. Lack of motivation from PD volunteers to help me which I will explain at the end 
9. Sometimes Toxic thoughts of trainees because of lack of management in cyf and affected them to commit to the module
10. Lack of space and this module needs to be in-person in my opinion.
11. Communication in Slack and ask people to do something with a text in Slack rather than assign them a task (this was not from me)

_**How we can make it easier for the next portfolio module Leaders:**_

1. Having a portfolio module in the syllabus like the way we have it for JS: 

- For example, having a plan for each week at least for mornings where we cover PD sessions and workshops to help volunteers and speed them up in finding the right people for their sessions. like having subjects and plans for mornings and the module lead needs to just ask volunteers who want to lead that subject and present it. 

We had these subjects for mornings and PD sessions: 
STAR method for soft skills and practice it with them, soft skills questions, how to be resilient and fear of rejection when they apply for jobs, different kinds of technical interviews and practice it with them, review their CVs and their Linkedins and show them all of the feature of Linkedins they can use or add, How to do networking and connect with employers

2. Maybe divide people into two different paths and ask them which path they want to go for:

- one path: build portfolio website, work on their CVs and do technical questions like coders(we can also think about some tasks/home works for them)
- second path: build projects in group or individual (I had people who wanted to work in the group and also people who did not want to work in the group but all of them wanted to make. a project in this portfolio)

In my opinion portfolio module has 2 parts to help trainees: soft skills + improving their LinkedIn, the CV part which is happening in the mornings and the second part is technical skills: building their project or portfolio.

3. Have a group of volunteers who can commit to this module for the technical part. for people who were creating a project in a group, it was beneficial for them to have at least one technical lead in their channel to review their PRs and help them with their questions.
4. Assign homework and tasks to trainees(technical + soft skills). I hate to admit that trainees work better when their work can be tracked and they have been asked to do something as their homework.
5. Have in-person sessions as much as you can.


**How we can help trainees- These can be suggestions for the Portfolio module syllabus**

- Do retro every session and ask trainees for their feedback at the end of the day. It is such an important part where you can ask them to be honest with you and help you with how your session went and how you can make it better for next week or the next portfolio module.
- Specific 15-20 minutes each session to ask them to think about their next week's goals based on what they achieved so far which will help them to be realistic. I would also add something here that I did not do: they should talk to each other about their next week availability and how much they can achieve as an individual.
- Follow-ups with trainees during the week - send a follow-up text every week, encourage them to keep going and ask them what they want to achieve till the end of the week.
- We should make sure to assign people based on their availability, their commitment and the stages they are in their journey.


**Feedback for CYF: Lack of motivation from PD volunteers to help me**

This is my feedback for CYF and I kept thinking about this when I was doing this module. 
@haldenb  mentioned to me a few times that trainees had different PD sessions about CV and LinkedIn before and also some of them have great PD volunteers who help some of them a lot to improve their CVs.

We have some amazing PD volunteers who are doing amazing jobs and they have their trainees from the start of the cohort to help them with CV and PD sessions.
Having a good and committed PD or Technical volunteer is by chance and some trainees are getting lots of support and some do not have the same support.
My new suggestion for the new cohort is we do not assign them PD volunteers and help them in this module or something similar so everyone has the same opportunity and the same support. either we want to extend this module to 5 weeks or more I think it is good to summarize everything in this module instead of keep telling them about portfolio websites or CVs from day one and them keep hearing it and ignoring it till the end of the cohort they want to apply for jobs.
Based on what I saw from most of the trainees their CVs and Linkedin were not good enough for applying for jobs even if they had PD volunteers, PD sessions, etc. We will save time for everyone and if we have any sessions it will be more efficient.

One feedback is we need to emphasise things when it is closer to time and it is important and maybe it is good to think about what is going wrong from CYF and think about the structure of modules.

Thanks

**Reply by SallyMcGrath** [reply]
Just to gently remind you that you are CYF. ❤️ It's not separate from you, Ellie.

Can I confirm here - nobody in London used the provided curriculum? Do you have an insight as to why?

**Comment by jo-hsuanc** [comment]
Hi all👋, I'm Jo, the UI/UX volunteer. Alec shared this discussion thread with a few design volunteers, and thought we could contribute our thoughts regarding our experience helping trainees with their projects.
(Disclaimer: I have only helped a group of trainees on one of their portfolio projects, so my comments might not reflect all truths. And I'm saying this from a designer's perspective, and I have no experience interviewing developers; I purely gave feedback based on the quality of their projects.)

Issues I've found are that their projects seem to lack proper product discovery, for example, _what's the problem or opportunity_, _who are the main users/customers_, and _how is the solution valid_? I understand this wouldn't be their focus when it comes to employability. However, when employers review their portfolio, it would be problematic if what they solved is not a pre-existing problem or if the solution simply didn't work.

Another side issue I realised is that design/product thinking is lacking during the process, and what we did for the trainees was actually 'reverse engineering.' They asked for our help after graduation, which I believe should have happened earlier. However, I did see the potential in their projects and did our best to rationalise their projects and use resources to make them look like 'industry-standard.'

All in all, I do believe trainees are motivated, but they might need a better understanding of how the industry works and what the benchmark is for quality products. We, as design volunteers, are always happy to support on product discovery and design execution, and we could expand our feedback sessions(slack: #design-feedback) a wider audience (originally, it's only for trainees/graduates who are interested in UI/UX and design volunteers).

Ping me on Slack or post messages in the #design-feedback channel if you want to discuss more in details. 🙌
